src/adam/core/rbd_algorithms.py

```diff
@@ -325,18 +366,25 @@ def CoM_position(
         for item in self.robot_desc.link_map:
             link = self.robot_desc.link_map[item]
             if link.inertial is not None:
-                T_fk = self.forward_kinematics(item, base_transform, joint_positions)
+                I, mass, o, rpy = self.extract_link_properties(link, density, length_multiplier)
+                T_fk = self.forward_kinematics(item, base_transform, joint_positions, density, length_multiplier)
                 T_link = self.H_from_Pos_RPY(
-                    link.inertial.origin.xyz,
-                    link.inertial.origin.rpy,
+                    o,
+                    rpy,
                 )
                 # Adding the link transform
-                T_fk = T_fk @ T_link.array
-                com_pos += T_fk[:3, 3] * link.inertial.mass
-        com_pos /= self.get_total_mass()
+                T_fk = T_fk @ T_link
```

Comment:
By comparing with the code without the modification, it seems that the code is meant to be:
```suggestion
                T_fk = T_fk @ T_link.array
```
with this modification, the test seems to pass.

Reply:
Thanks @traversaro , I had completely missed that modification !